src/imputation/imputation_main.py

```diff
@@ -61,48 +69,83 @@ def run_imputation(
     df.loc[clear_responders_mask, "imp_marker"] = "R"
     df.loc[~clear_responders_mask, "imp_marker"] = "no_imputation"
 
+    # Create an 'instance' of value 1 for non-responders and refs with 'No R&D'
+    df = hlp.instance_fix(df)
+    df = hlp.duplicate_rows(df)
+
     # remove records that have had construction applied before imputation
     if "is_constructed" in df.columns:
-        constructed_df = df.copy().loc[df["is_constructed"].isin([True]) & df["force_imputation"].isin([False])]
+        constructed_df = df.copy().loc[
```

Comment:
Where do we bring the constructed_df back in before estimation/outputs

Reply:
at the end of imputation main, line 119